sdk/src/activity_context.rs

```diff
@@ -0,0 +1,212 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{Duration as StdDuration, SystemTime};
+
+use prost_types::{Duration, Timestamp};
+use temporal_sdk_core_api::Worker;
+use temporal_sdk_core_protos::coresdk::{
+    activity_task,
+    common::{Payload, RetryPolicy, WorkflowExecution},
+    ActivityHeartbeat,
+};
+use tokio_util::sync::CancellationToken;
+
+/// Used within activities to get info, heartbeat management etc.
+#[derive(Clone)]
+#[allow(dead_code)]
+pub struct ActContext {
+    pub(crate) worker: Arc<dyn Worker>,
+    pub(crate) cancellation_token: CancellationToken,
+    pub(crate) input: Vec<Payload>,
+    pub(crate) heartbeat_details: Vec<Payload>,
+    /// #NOTE future usage?
+    pub(crate) header_fields: HashMap<String, Payload>,
+    pub(crate) info: ActivityInfo,
+}
+
+#[derive(Clone)]
+pub struct ActivityInfo {
+    pub task_token: Vec<u8>,
+    pub workflow_type: String,
+    pub workflow_namespace: String,
+    pub workflow_execution: Option<WorkflowExecution>,
+    pub activity_id: String,
+    pub activity_type: String,
+    pub task_queue: String,
+    pub heartbeat_timeout: Option<Duration>,
+    /// Time of activity scheduled by a workflow
+    pub scheduled_time: Option<Timestamp>,
+    /// Time of activity start
+    pub started_time: Option<Timestamp>,
+    /// Time of activity timeout
+    pub deadline: Option<Timestamp>,
+    /// Attempt starts from 1, and increased by 1 for every retry if retry policy is specified.
+    pub attempt: u32,
+
+    /// #NOTE Not in sdk-go
```

Comment:
Not super important as a docstring. Feel free to either drop it and I'll add one later, or give it a description.